src/control/server/ctl_storage_rpc.go

```diff
@@ -340,17 +340,58 @@ func (c *ControlService) StorageFormat(ctx context.Context, req *ctlpb.StorageFo
 	resp.Mrets = make([]*ctlpb.ScmMountResult, 0, len(instances))
 	resp.Crets = make([]*ctlpb.NvmeControllerResult, 0, len(instances))
 	scmChan := make(chan *ctlpb.ScmMountResult, len(instances))
+	mdFormatted := false
+
+	// Format control metadata first, if needed
+	if needs, err := c.storage.ControlMetadataNeedsFormat(); err != nil {
```

Comment:
I haven't actually tested this yet. I will try it.

Reply:
If someone has a `control_metadata` field in their config, it will required a control metadata format if the directory structure is missing. If they're using a tmpfs or pmem config, the behavior remains the same.

Do you think this is acceptable for the remaining users who use the recreate superblocks flag? I'm not sure I want to put in the effort to make new functionality support a deprecated option.

Reply:
I don't use the flag myself so I'm not completely sure but we will hear some noise from those who do use it if it is broken. I think it's acceptable given it's a deprecated option.